src/main/resources/functions/adGroupAdsMutate.xml

```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ ~  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ ~
+ ~  WSO2 LLC. licenses this file to you under the Apache License,
+ ~  Version 2.0 (the "License"); you may not use this file except
+ ~  in compliance with the License.
+ ~  You may obtain a copy of the License at
+ ~
+ ~    http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~  Unless required by applicable law or agreed to in writing,
+ ~  software distributed under the License is distributed on an
+ ~  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ ~  KIND, either express or implied.  See the License for the
+ ~  specific language governing permissions and limitations
+ ~  under the License.
+-->
+
+<!-- Generated on 23-Fri, 08, 2024 16:11:19+0530 -->
+
+<template xmlns="http://ws.apache.org/ns/synapse" name="adGroupAdsMutate">
+    <!-- Path Parameter List -->
+    <parameter name="customerId" description="Required. The ID of the customer whose ads are being modified."/>
+    <!-- Request Body Parameter List -->
+    <parameter name="operations"
+               description="Required. The list of operations to perform on individual ads."/>
+    <parameter name="partialFailure"
+               description="If true, successful operations will be carried out and invalid operations will return errors. If false, all operations will be carried out in one transaction if and only if they are all valid. Default is false."/>
+    <parameter name="validateOnly"
+               description="If true, the request is validated but not executed. Only errors are returned, not results."/>
+    <parameter name="responseContentType"
+               description="The response content type setting. Determines whether the mutable resource or just the resource name should be returned post mutation."/>
+    <sequence>
+        <class name="org.wso2.carbon.google.ads.connector.RestURLBuilder">
+            <property name="operationPath" value="/v17/customers/{customerId}/adGroupAds:mutate"/>
+            <property name="pathParameters" value="customerId,"/>
+        </class>
+        <payloadFactory media-type="json" template-type="freemarker">
+            <format>
+                <![CDATA[{
+                                                                                        <#if (args.arg1)?has_content>
+                                    "operations": ${args.arg1},                                </#if>
```

Comment:
Can use following to mitigate optional args causing an invalid json. If the json being created has to be a nested one, use typeching and then dump the map. Update the gen tool accordingly

```ftl
<#assign argMap = {}>

<#if (args.arg1)?has_content>
  <#assign argMap = argMap + {"country": args.arg1}>
</#if>

<#if (args.arg2)?has_content>
  <#assign argMap = argMap + {"city": args.arg2}>
</#if>

<#if (args.arg3)?has_content>
  <#assign argMap = argMap + {"city": args.arg3}>
</#if>


{
<#list argMap as k,v>
 "${k}" : "${v}"<#sep>,
</#list>
}
```

Reply:
@arunans23 Please incorporate this improvement when designing the new gen tool.